hal/halx86/smp/smp.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Core source file for SMP management
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include <hal.h>
+#include <smp.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+extern PPROCESSOR_IDENTITY HalpProcessorIdentity;
+
+/* FUNCTIONS *****************************************************************/
+
+VOID
+HalpSetupProcessorsTable(UINT32 NTProcessorNumber)
+{
+    PKPRCB CurrentPrcb;
+
+    /* 
+     * Link the Prcb of the current CPU to
+     * the current CPUs entry in the global ProcessorIdentity 
+     */
+    CurrentPrcb = KeGetCurrentPrcb();
```

Comment:
Do you need this variable for some future code in this function? (If not, assignation below could directly be done with `KeGetCurrentPrcb()`.)

Reply:
My intention is to expand this with some other stuff, also if we ever go into SMT / x2APIC
it'll be expanded quite a bit.